### PR TITLE
Include `_call_params: CallParams` attribute in `Prompt`

### PR DESCRIPTION
**Pull request by willbakst**
- This inclusion enables co-locating all aspects of the call to the LLM that may impact the quality of the `Prompt`.

### CODE REVIEW

File: mirascope/chat/models/async_openai_chat.py [hunk 1]
```diff
@@ -140,6 +162,27 @@ async def stream(
             OpenAIError: raises any OpenAI errors, see:
                 https://platform.openai.com/docs/guides/error-codes/api-errors
         """
+        if isinstance(prompt, Prompt):
+            if self.model != "gpt-3.5-turbo":
```
Comment (brenkao) [inline mirascope/chat/models/async_openai_chat.py:166]: ditto

File: mirascope/chat/models/openai_chat.py
Comment (brenkao) [inline mirascope/chat/models/openai_chat.py:1]: same as async

**Review by brenkao — APPROVED** [review]

File: mirascope/prompts.py [hunk 2]
```diff
@@ -26,6 +32,36 @@ def _format_template(prompt: Prompt, template: str) -> str:
     return template.format(**{var: getattr(prompt, var) for var in template_vars})
 
 
+class CallParams(BaseModel):
```
Comment (brenkao) [inline mirascope/prompts.py:35]: does this limit us when we implement other models. feels like prompt should take in a union of different call params and the first one we support is OpenAICallParams

Comment (willbakst) [reply mirascope/prompts.py]: Good call. Updated

**Pull request merged**